rpcserver.go

```diff
@@ -1199,14 +1199,40 @@ func (r *rpcServer) SendPayment(paymentStream lnrpc.Lightning_SendPaymentServer)
 		case err := <-errChan:
 			return err
 		case nextPayment := <-payChan:
-			// Parse the details of the payment which include the
-			// pubkey of the destination and the payment amount.
-			dest := nextPayment.Dest
-			amt := btcutil.Amount(nextPayment.Amt)
-			destNode, err := btcec.ParsePubKey(dest, btcec.S256())
-			if err != nil {
-				return err
 
+			// Construct a payment request to send to the
```

Comment:
See my comment above the in the `cmd/lncli` package: a new RPC should be added rather than modifying the existing one.